update_repo_properties.py

```diff
@@ -0,0 +1,240 @@
+# Parse the same information as in the branch protection, but format RestAPI calls
+# Currently github does not have graphql support for custom properties
+# https://docs.github.com/en/rest/repos/custom-properties?apiVersion=2022-11-28
+
+
+from __future__ import annotations
+
+import argparse
+import csv
+from dataclasses import dataclass, fields
+from enum import Enum
+from pathlib import Path
+from typing import List, Optional
+
+# need this new fangled thinger for rest API
+# needs a fine grained PAT to work
+from ghapi.all import GhApi
+
+api = None
+
+
+def get_gh_api() -> GhApi:
+    global api
+    if api is None:
+        api = GhApi()
+
+    return api
+
+
+class RepoType(str, Enum):
+    """Enum where string values match the exact property values on Github"""
+    NONE = 'None'
+    PYTHON_LIBRARY = 'Python Library'
+    TWINCAT_LIBRARY = 'TwinCAT Library'
+    PLC = 'PLC'
+    BACKUP = 'Backup'
+    EPICS_IOC = 'EPICS IOC'
+    EXTERNAL = 'External'
+    PYTHON_DEV = 'Python Dev'
+    EPICS_MODULE = 'EPICS module'
+    OTHER = 'Other'
+
+
+class StatusChecks(str, Enum):
+    """Enum where string values match the exact property values on Github"""
+    PYTHON_STANDARD = 'Python Standard'
+    TWINCAT_STANDARD = 'TwinCAT Standard'
+    NONE = 'None'
+
+
+@dataclass
+class RepoProperty:
+    """field names match table headers exactly"""
+    repo_name: str
+    owner: str = 'pcdshub'
+    repo_type: Optional[RepoType] = None
+    default: Optional[bool] = None
+    master: Optional[bool] = None
+    gh_pages: Optional[bool] = None
+    status_checks: Optional[StatusChecks] = None
```

Comment:
Do we need to update this script if we change our property names?
Do we need to update this script if another group in the same org adds their own property names?

Reply:
We will have to update this if we change our property names, yes.

If other groups in the org add their own properties, I imagine we'll just have to agree not to change each other's props. As it is the script will just change the ones requested.

We'll probably have to change the names to something prepended with "ecs_", but that's a bridge we can cross later